lib/pengine/native.c

```diff
@@ -599,6 +599,17 @@ pcmk__native_output_string(pe_resource_t *rsc, const char *name, pe_node_t *node
         g_string_append_printf(outstr, " %s", node->details->uname);
     }
 
+    // Failed probe operation
+    if (native_displayable_role(rsc) == RSC_ROLE_STOPPED) {
+        xmlNode *probe_op = pe__failed_probe_for_rsc(rsc, node ? node->details->uname : NULL);
```

Comment:
Well at the least, this function can wipe out node:
```
    if (pcmk_is_set(show_opts, pcmk_show_rsc_only)
        || pcmk__list_of_multiple(rsc->running_on)) { 
        node = NULL; 
    }
```

Reply:
Ah yes, that's used when resources are being listed for a particular node only (in which case there's no need to list the node again), or for the cluster as a whole (crm_resource --list). In that case, we do need to check on the original node that was passed.

I believe node can also be NULL for some stopped resources. In that case I don't think there's a clear best approach. Someone might expect it to show "(not installed)" only if that's the case on all nodes, or maybe not. I'm thinking the best thing to do (if the original node is NULL) is to not show any extended reason, since the reason might be different on different nodes.

Reply:
An interesting thing to note... if I add `node != NULL &&` to the condition on line 603 in the patch above, it gets rid of everywhere that the "(not installed)" string or other extended reason was shown in the updated test cases.

Reply:
Yep -- stopped resources are generally not shown with a node name, since by definition they have no node. There are exceptions like shutdown-locked resources and partially active resources.

I'm thinking pe__failed_probe_for_rsc() should return NULL if node is NULL. If we don't know the node, we can't be sure a failed probe entry is relevant.

Reply:
That has the same effect on the test cases, which makes me start to wonder when exactly those extended reasons should be getting shown.

Reply:
Bah :(

I guess we do need to think about it more. For:

      * Full List of Resources:
        * Fencing	(stonith:fence_xvm):	 Started cluster01
        * dummy-1	(ocf:pacemaker:Dummy):	 Stopped (not installed) 

It doesn't make sense to display "not installed" unless it's not installed on every node. We still want the scheduler to treat the failure as not installed on the nodes where that's the case, but for display purposes, "Stopped" is probably sufficient.

Similarly, for:

        * Clone Set: ping-1-clone [ping-1]:
          * Stopped (not installed): [ cluster01 cluster02 ]

It only makes sense if it's not installed on every node in the stopped list.

It does make sense to say "not installed" for a particular node when we're listing by node. So maybe "return NULL if node is NULL" is a reasonable approach, and we'd have to list by node to see any difference.

Reply:
The scheduler handling is the most important part -- treating not-installed results as not-running.

Reply:
And not listing them in the failed actions section